Add pre-made Activities for discord Rich Presence

YodaForce157 opened the pull request:
Created a class to update the Discord Rich Presence with either a predefined activity or a custom one.

**Tasks for tomorrow**

- [x] Make the Activities look nice
- [x] Probably remove over the top details like - Selecting a project etc, primarily focus on editing projects etc
- [x] Whatever else comes to mind
- [x] Question reality
- [x] Do anything that either pops to mind or I'm asked to do

Commit:
Create RailroadActivities.java

Create railroadActivites with an enum with each type of activity, also make the create project & back button update the activity

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Remove unnecessary activities

Removed unnecessary activities and updated to latest commit, if there are any issues with the pr it's likely something I did because the merge conflict menu confuses me

Commit:
Update RailroadActivities.java

Commit:
Fix conflicts

Commit:
Remove pre-fetch comments

DaRealTurtyWurty commented on src/main/java/io/github/railroad/Railroad.java:
```diff
@@ -63,20 +62,14 @@ private static void handleStyles(Scene scene) {
+        
     }
 
+
```
```suggestion
```

DaRealTurtyWurty commented on src/main/java/io/github/railroad/Railroad.java:
```diff
@@ -63,20 +62,14 @@ private static void handleStyles(Scene scene) {
 
         String baseTheme = getResource("styles/base.css").toExternalForm();
         scene.getStylesheets().add(baseTheme);
+        
```
```suggestion
```

DaRealTurtyWurty commented on src/main/java/io/github/railroad/discord/activity/RailroadActivities.java:
```diff
@@ -0,0 +1,37 @@
+    public enum RailroadActivityTypes{
+        RAILROAD_DEFAULT(),
+        EDIT_FILE(),
+
```
```suggestion
```

DaRealTurtyWurty commented on src/main/java/io/github/railroad/discord/activity/RailroadActivities.java:
```diff
@@ -0,0 +1,37 @@
+public class RailroadActivities {
+    public enum RailroadActivityTypes{
+        RAILROAD_DEFAULT(),
+        EDIT_FILE(),
```
```suggestion
        RAILROAD_DEFAULT,
        EDIT_FILE;
```

DaRealTurtyWurty commented on src/main/java/io/github/railroad/discord/activity/RailroadActivities.java:
```diff
@@ -0,0 +1,37 @@
+    public static void setActivity(DiscordActivity activity){
+        Railroad.getDiscord().getActivityManager().updateActivity(activity);
+    }
+    public static void setActivity(RailroadActivityTypes type){
```
```suggestion
    }
    
    public static void setActivity(RailroadActivityTypes type) {
```

DaRealTurtyWurty commented on src/main/java/io/github/railroad/discord/activity/RailroadActivities.java:
```diff
@@ -0,0 +1,37 @@
+
+    }
+
+    public static void setActivity(DiscordActivity activity){
```
```suggestion
    public static void setActivity(DiscordActivity activity) {
```

DaRealTurtyWurty commented on src/main/java/io/github/railroad/discord/activity/RailroadActivities.java:
```diff
@@ -0,0 +1,37 @@
+    public static void setActivity(RailroadActivityTypes type){
+        //TODO add current project name & file name into details
+        var activity = new DiscordActivity();
+        switch(type){
```
```suggestion
        switch(type) {
```

DaRealTurtyWurty commented on src/main/java/io/github/railroad/discord/activity/RailroadActivities.java:
```diff
@@ -0,0 +1,37 @@
+                activity.getTimestamps().setStart(Instant.now());
+                activity.getAssets().setLargeImage("logo");
+        }
+        Railroad.getDiscord().getActivityManager().updateActivity(activity);
```
```suggestion
        }
        
        Railroad.getDiscord().getActivityManager().updateActivity(activity);
```

DaRealTurtyWurty commented on src/main/java/io/github/railroad/discord/activity/RailroadActivities.java:
```diff
@@ -0,0 +1,37 @@
+                activity.setState("v0.1.0");
+                activity.getTimestamps().setStart(Instant.now());
+                activity.getAssets().setLargeImage("logo");
+        }
```
Would be a good idea to add a default case to this. I know intellij will complain saying its not needed, but its more of a future protection thing rather than something that affects the current code.

DaRealTurtyWurty reviewed (changes requested):
Mostly just whitespace changes but a few other minor tweaks too and this will be good to go. 👍

Commit:
Apply suggestions and edit details to be more generic for default & railroad_default

Honestly unsure if having RAILROAD_DEFAULT and default being the same is a good idea or not, but it's like that so, feel free to change it.

YodaForce157 commented:
All sorted.

DaRealTurtyWurty commented:
> All sorted.

You missed one
https://github.com/Railroad-Team/Railroad/pull/8/files/d2acb66f4aca623c22c482b61da2f6f30eb4e8cb#diff-0c64c254bd975ec6bfb2c18c0557e2803e8348ea3a3f828619f315107eb2f4dc

YodaForce157 commented:
There, latest commit sorted it, just missed that one when clicking resolve issue

Commit:
Update RailroadActivities.java
